### PR TITLE
treewide: fix typos found by 'codespell'

### PR DESCRIPTION
**Pull request by rohieb**

### CODE REVIEW

File: docs/integration.rst [hunk 1]
```diff
@@ -400,7 +400,7 @@ provides a dedicated boot source selection framework, called *bootchooser*,
 backed by an atomic and redundant storage backend, named *state*.
 
 *Barebox state* allows you to save the variables required by bootchooser with
-memory specific storage strategies in all common storage medias,
+memory specific storage strategies in all common storage mediums,
```
Comment (a3f) [inline docs/integration.rst:403]: Shouldn't that be media?

Comment (rohieb) [reply docs/integration.rst:403]: [Both are correct](https://en.wiktionary.org/wiki/medium#Noun).

**Review by jluebbe — APPROVED** [review]

**Comment by jluebbe** [comment]
I fixed an additional spelling error and added a tests step via github actions.

**Review by jluebbe — APPROVED** [review]

**Review by ejoerns — APPROVED** [review]
Amazing that it does not seem to have any false-positives.

**Pull request merged**